### DISCUSSION TITLE
getQueryData & Selectors.

### DISCUSSION THREAD

**Discussion by OmerCohenAviv** [TanStack/query#4093] [Q&A] [answered]
Hey, I was wondering about getQueryData with selectors,
As getQueryData returns the data without any transformations,

Is it possible to target selectors with getQueryData ?

it could be nice to have a hook **getQuerySelector(queryKey, selectorKey)** 🤔 

or I should just use selectors like mention here https://tkdodo.eu/blog/react-query-data-transformations#3-using-the-select-option
Thanks!

**Comment by TkDodo** [comment, marked as answer]
`getQueryData` gets the data from the cache. There is only one structure in the cache. The `select` option is "per observer", so every useQuery call can transform data like they see fit, and additionally, only subscribe to the transformation result.

`getQueryData` doesn't create a subscription, so a selector is unnecessary. You can just do:

```
const cachedData = queryClient.getQueyData(...)
const transformedData = transformData(cachedData)
```
